### PR TITLE
Remove unnecessary stages in Jenkin's file

### PR DESCRIPTION
**Pull request by ahoffer**
Removed several stages from Jenkins that should no longer be necessary.
Removed builds stages and static analysis stages because Circle CI now handles them.
Removed schedule for nightly builds of master.

### CODE REVIEW

**Comment by ahoffer** [comment]
I'm not sure how to test this outside of actually creating a release.

File: Jenkinsfile [hunk 1]
```diff
@@ -33,13 +33,6 @@ pipeline {
         disableConcurrentBuilds()
         timestamps()
     }
-    triggers {
-        /*
-          Restrict nightly builds to master branch, all others will be built on change only.
-          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
-        */
-        cron(env.BRANCH_NAME == "master" ? "H H(21-23) * * *" : "")
-    }
```
Comment (emmberk) [inline Jenkinsfile]: ✏️ Delete the PR comment stuff in the setup block

File: Jenkinsfile [hunk 2]
```diff
@@ -87,74 +80,7 @@ pipeline {
...
-            }
-        }
-        stage('Full Build') {
-            when { expression { env.CHANGE_ID == null } }
-            parallel {
-                stage ('Linux') {
-                    steps {
-                        withMaven(maven: 'Maven 3.5.4', jdk: 'jdk11', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'cx-internal', mavenOpts: '${LINUX_MVN_RANDOM}') {
-                            script {
-                                if (params.RELEASE == true) {
```
Comment (emmberk) [inline Jenkinsfile]: ❗️Keep this section

File: Jenkinsfile [hunk 3]
```diff
@@ -87,74 +80,7 @@ pipeline {
...
-                    when {
-                        allOf {
-                            expression { env.CHANGE_ID == null }
-                            branch 'master'
-                            environment name: 'JENKINS_ENV', value: 'prod'
-                        }
-                    }
-                    steps {
-                        script {
-                            if (params.RELEASE == true) {
```
Comment (emmberk) [inline Jenkinsfile]: ❗️Keep this section

File: Jenkinsfile (hunk 1 above)
Comment (emmberk) [inline Jenkinsfile]: Reminder to remove the PR configuration on Jenkins:
image

Comment (ahoffer) [reply Jenkinsfile]: Done

File: Jenkinsfile [hunk 4]
```diff
@@ -33,77 +33,43 @@ pipeline {
...
-                                //  Check to see if there are multiple parents for the commit. (merged)
-                                sh(script: 'if [ `git cat-file -p HEAD | head -n 3 | grep parent | wc -l` -gt 1 ]; then exit 1; else exit 0; fi')
-                                //  No error was thrown -> we called exit 0 -> HEAD is not a merge commit/doesn't have multiple parents
-                                env.PR_COMMIT = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
-                            } catch (err) {
-                                //  An error was thrown -> we called exit 1 -> HEAD is a merge commit/has multiple parents
-                                env.PR_COMMIT = sh(returnStdout: true, script: 'git rev-parse HEAD~1').trim()
-                            }
+                    //  Clear existing status checks
+                    def jsonBlob = getGithubStatusJsonBlob("pending", "${BUILD_URL}display/redirect", "Full Build In Progress...", "CX Jenkins/Full Build")
```
Comment (emmberk) [inline Jenkinsfile]: ✏️ Remove these 2 lines

File: Jenkinsfile [hunk 5]
```diff
@@ -198,39 +164,7 @@ pipeline {
...
-        }
-        unstable {
-            slackSend color: '#ffb600', message: "UNSTABLE: ${JOB_NAME} ${BUILD_NUMBER}. See the results here: ${BUILD_URL}"
-            withCredentials([usernameColonPassword(credentialsId: 'cxbot', variable: 'GITHUB_TOKEN')]) {
-                postCommentIfPR("⚠️ Build unstable. See the job results in [legacy Jenkins UI](${BUILD_URL}) or in [Blue Ocean UI](${BUILD_URL}display/redirect).", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
-                script {
-                    def jsonBlob = getGithubStatusJsonBlob("failure", "${BUILD_URL}display/redirect", "Full build was unstable!", "CX Jenkins/Full Build")
-                    postStatusToHash("${jsonBlob}", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${env.PR_COMMIT}", "${GITHUB_TOKEN}")
-                }
+                }d
```
Comment (emmberk) [inline Jenkinsfile]: ✏️ typo

**Comment by ahoffer** [comment]
Build now (should do nothing)

File: Jenkinsfile (hunk 5 above)
Comment (ahoffer) [reply Jenkinsfile]: Fixed

File: Jenkinsfile (hunk 4 above)
Comment (ahoffer) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 1 above)
Comment (ahoffer) [reply Jenkinsfile]: I reviewed and the setup block and I don't see anything in their about PRs, so we should be good now.

**Pull request merged**